src/EnergyPlus/HeatPumpWaterToWaterSimple.cc

```diff
@@ -123,159 +125,113 @@ namespace HeatPumpWaterToWaterSimple {
-                }
+        } else if (this->WWHPPlantTypeOfNum == DataPlant::TypeOf_HPWaterEFHeating) {
+            if (calledFromLocation.loopNum == this->LoadLoopNum) { // chilled water loop
+                this->InitWatertoWaterHP(this->WWHPPlantTypeOfNum, this->Name, FirstHVACIteration, CurLoad);
+                this->CalcWatertoWaterHPHeating(CurLoad);
+                this->UpdateGSHPRecords();
+            } else if (calledFromLocation.loopNum == this->SourceLoopNum) { // condenser loop
+                PlantUtilities::UpdateChillerComponentCondenserSide(this->SourceLoopNum,
+                                                                    this->SourceLoopSideNum,
+                                                                    DataPlant::TypeOf_HPWaterEFHeating,
+                                                                    this->SourceSideInletNodeNum,
+                                                                    this->SourceSideOutletNodeNum,
+                                                                    -this->reportQSource,
+                                                                    this->reportSourceSideInletTemp,
+                                                                    this->reportSourceSideOutletTemp,
+                                                                    this->reportSourceSideMassFlowRate,
+                                                                    FirstHVACIteration);
             } else {
-                ShowFatalError("SimHPWatertoWaterSimple:: Invalid " + HPEqFitHeating + ", Requested Unit=" + GSHPName);
+                ShowFatalError("SimHPWatertoWaterSimple:: Invalid loop connection " + HPEqFitCooling + ", Requested Unit=" + this->Name);
             }
         } else {
-            ShowFatalError("SimHPWatertoWaterSimple: Module called with incorrect GSHPType=" + GSHPType);
+            ShowFatalError("SimHPWatertoWaterSimple: Module called with incorrect GSHPType");
         } // TypeOfEquip
     }
 
-    void GetWatertoWaterHPInput()
+    void GshpSpecs::getDesignCapacities(const PlantLocation &calledFromLocation, Real64 &MaxLoad, Real64 &MinLoad, Real64 &OptLoad)
+    {
+        this->InitWatertoWaterHP(
+            this->WWHPPlantTypeOfNum, this->Name, true, 0.0); // Changed to true for FirstHVAC and 0.0 for Curload during this phase
+        if (calledFromLocation.loopNum == this->LoadLoopNum) {
+            if (this->WWHPPlantTypeOfNum == DataPlant::TypeOf_HPWaterEFCooling) {
+                this->sizeCoolingWaterToWaterHP();
+                MinLoad = 0.0;
+                MaxLoad = this->RatedCapCool;
+                OptLoad = this->RatedCapCool;
+            } else if (this->WWHPPlantTypeOfNum == DataPlant::TypeOf_HPWaterEFHeating) {
+                this->sizeHeatingWaterToWaterHP();
+                MinLoad = 0.0;
+                MaxLoad = this->RatedCapHeat;
+                OptLoad = this->RatedCapHeat;
+            } else {
```

Comment:
These coils look basically identical. The capacity could be just RatedCap and they would be much more similar.

Reply:
Absolutely right! I will be reducing it back down as much as possible, but wanted to get it fleshed out to get my little mocked up unit tests working.  I was actually debating on reducing it to one single input object that includes a cooling/heating choice as well but will wait to hear some review feedback before going down that road.

Reply:
Keep the 2 objects for now and reduce if you can. I understand the fleshing and am hesitant to say anything while I'm in the middle of UnitarySystem (you could say the same thing to me).